### PR TITLE
Add java dump end date and approximate duration

### PR DESCRIPTION
**Pull request by kgibm**
Sometimes it's useful to know the approximate time it takes to produce java dumps to gauge their overhead:

```
0SECTION       Javadump End section
NULL           =================================
1TIDMPCOMPLETE Dump completed: 2022/09/26 at 14:04:58:585
1TIDMPDURATION Approximate time to produce this dump: 2 ms
NULL           ---------------------- END OF DUMP -------------------------------------
```

Signed-off-by: Kevin Grigorenko <kevin.grigorenko@us.ibm.com>

### CODE REVIEW

File: runtime/rasdump/javadump.cpp [hunk 1]
```diff
@@ -3378,6 +3381,32 @@ JavaCoreDumpWriter::writeTrailer(void)
...
+
+	_OutputStream.writeCharacters("1TIDMPCOMPLETE Dump Completed Date: ");
+	_OutputStream.writeCharacters(timeStamp);
+	_OutputStream.writeInteger(now % 1000, ":%03d"); /* Add the milliseconds. */
+	_OutputStream.writeCharacters("\n");
+
+	/* Calculate the approximate time it took to produce the dump. */
+	int64_t duration = now - _DumpStart;
+
+	_OutputStream.writeCharacters("1TIDMPDURATION Approximate time to produce the dump (ms): ");
```
Comment (keithc-ca) [inline runtime/rasdump/javadump.cpp:3405]: I don't think we need to capitalize each word in the `1TIDMPCOMPLETE` line, just "Dump", for consistency with `1TIDMPDURATION`. Also, I think the `(ms)` before the time should be removed, and replaced by `ms` after it. Here's a sample of what I think the output should be:
```
1TIDMPCOMPLETE Dump completed: 2022/09/26 at 14:40:47:718
1TIDMPDURATION Approximate time to produce this dump: 4 ms
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by keithc-ca** [comment]
Looks good: please squash.

**Comment by kgibm** [comment]
Squashed. Thanks!

**Comment by keithc-ca** [comment]
jenkins test sanity win,win32 jdk8

**Comment by keithc-ca** [comment]
32-bit problem:
```
01:48:12.443  F:\Users\jenkins\workspace\Build_JDK8_x86-32_windows_Personal\openj9\runtime\rasdump\javadump.cpp(3402): error C2220: warning treated as error - no 'object' file generated
01:48:12.443  F:\Users\jenkins\workspace\Build_JDK8_x86-32_windows_Personal\openj9\runtime\rasdump\javadump.cpp(3402): warning C4244: 'argument': conversion from 'int64_t' to 'UDATA', possible loss of data
```

**Comment by kgibm** [comment]
Added a commit to use `_OutputStream.writeInteger64` if you think that's a good solution.

File: runtime/rasdump/javadump.cpp [hunk 2]
```diff
@@ -3378,6 +3377,32 @@ JavaCoreDumpWriter::writeTrailer(void)
...
+	omrstr_ftime_ex(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, now, OMRSTR_FTIME_FLAG_LOCAL);
+	timeStamp[_MaximumTimeStampLength] = '\0';
+
+	_OutputStream.writeCharacters("1TIDMPCOMPLETE Dump completed: ");
+	_OutputStream.writeCharacters(timeStamp);
+	_OutputStream.writeInteger(now % 1000, ":%03d"); /* Add the milliseconds. */
+	_OutputStream.writeCharacters("\n");
+
+	/* Calculate the approximate time it took to produce the dump. */
+	int64_t duration = now - _DumpStart;
```
Comment (keithc-ca) [inline runtime/rasdump/javadump.cpp:3399]: I suggest this should use `uint64_t` (or `U_64`) with an explicit cast, for compatibility with its use in the call to `writeInteger64()`:
```
	uint64_t duration = (uint64_t)(now - _DumpStart);
```

Comment (kgibm) [reply runtime/rasdump/javadump.cpp]: Thanks, commit added

**Comment by kgibm** [comment]
Squashed

**Comment by keithc-ca** [comment]
jenkins test sanity win32 jdk8

**Comment by keithc-ca** [comment]
The 64-bit Windows build should be valid with the recent changes: https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK8_x86-64_windows/1308/

**Comment by pshipton** [comment]
FYI https://github.com/eclipse-openj9/openj9/issues/15976

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**